javascript/ql/src/semmle/javascript/frameworks/jQuery.qll

```diff
@@ -544,17 +544,17 @@ module JQuery {
   }
 
   /** A source of jQuery objects from the AST-based `JQueryObject` class. */
-  private DataFlow::Node legacyObjectSource() { result = any(JQueryObjectInternal e).flow() }
+  private DataFlow::SourceNode legacyObjectSource() {
+    result = any(JQueryObjectInternal e).flow().getALocalSource()
+  }
 
   /** Gets a source of jQuery objects. */
   private DataFlow::SourceNode objectSource(DataFlow::TypeTracker t) {
     t.start() and
     result instanceof ObjectSource::Range
     or
-    exists(DataFlow::TypeTracker init |
-      init.start() and
-      t = init.smallstep(legacyObjectSource(), result)
-    )
+    t.start() and
+    result = legacyObjectSource()
```

Comment:
Could you update the comment on line 594 and the code following it:
```codeql
// Handle contributed JQuery objects that aren't source nodes (usually parameter uses)
```
That was the reason it worked this way in the first place, but as of https://github.com/github/codeql/pull/3508, this is no longer necessary. I believe both the comment and the following is now out of date. (And also the legacy jQuery API has been deprecated long enough that we can remove/change it if we want).

Reply:
I think it's handled by `objectRef().getAMemberCall(name)` now.   
So those lines are just redundant.  

I've removed them.